lib/ansible/modules/rpm_key.py

```diff
@@ -151,7 +151,7 @@ def __init__(self, module):
     def fetch_key(self, url):
         """Downloads a key from url, returns a valid path to a gpg key"""
         rsp, info = fetch_url(self.module, url)
-        if info['status'] != 200:
+        if info['status'] != 200 or (url.startswith("ftp://") and info['status'] is None):
```

Comment:
Please create a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment). See [this fragment](https://github.com/ansible/ansible/blob/c2f528b7686c9573aa8493357219e0bbe4416dd3/changelogs/fragments/reboot-change-default-boot-command.yaml) as an example. Also, test for this change.

Reply:
Add a note in documentation section as well, saying that `key` can be an HTTP or a FTP URL.

Reply:
Changelog added and documentation section updated!